it/uk/gov/hmrc/transitmovements/repositories/MovementsRepositorySpec.scala

```diff
@@ -624,4 +622,147 @@ class MovementsRepositorySpec
 
   }
 
+  "updateMovement" should "return an error if there is no matching movement with the given ID" in forAll(
+    arbitrary[MovementId],
+    arbitrary[EORINumber],
+    arbitrary[MovementReferenceNumber]
+  ) {
+    (movementId, eori, mrn) =>
+      val message =
+        arbitrary[Message].sample.value.copy(body = None, messageType = MessageType.DepartureOfficeRejection, triggerId = Some(MessageId(movementId.value)))
+
+      val result = await(
+        repository.updateMovement(movementId, Some(eori), Some(mrn), instant).value
+      )
+
+      result should be(Left(MongoError.DocumentNotFound(s"No movement found with the given id: ${movementId.value}")))
+  }
+
+  it should "update the EORI if it is provided" in {
+    val departureMovement =
+      arbitrary[Movement].sample.value
+        .copy(
+          created = instant,
+          updated = instant,
+          messages = Vector(),
+          movementType = MovementType.Departure,
+          movementEORINumber = None,
+          movementReferenceNumber = None
+        )
+
+    await(
+      repository.insert(departureMovement).value
+    )
+
+    val result = await(
+      repository.updateMovement(departureMovement._id, Some(departureMovement.enrollmentEORINumber), None, receivedInstant).value
+    )
+
+    result should be(Right(()))
+
+    val movement = await {
+      repository.collection.find(Filters.eq("_id", departureMovement._id.value)).first().toFuture()
+    }
+
+    movement.updated shouldEqual receivedInstant
+    movement.movementEORINumber shouldEqual Some(movement.enrollmentEORINumber)
+    movement.movementReferenceNumber shouldEqual None
+  }
+
+  it should "update the MRN if they are both supplied" in {
```

Comment:
The description here got mixed up with the test below, should just be "update the MRN if it is provided" right